### ISSUE TITLE
Toggling menu with a game controller does not work as intended

### ISSUE THREAD

**Issue by jstolarek** [libretro/RetroArch#10776]
I have Menu Toggle button set to F1 and my gamepad's Home button (in Input > Hotkey Binds > Menu Toggle). The F1 button works as a toggle just as expected: when pressed in-game it goes into RA menu, when pressed in the menu it returns to the game. However, the gamepad Menu Toggle button only allows to enter the menu. When I press the menu Toggle pad button in the menu it does not return to the game. Happens with several different pads, all of them configured using autoconfig with an entry that looks like this:

```
input_menu_toggle_btn = "8"
input_menu_toggle_btn_label = "PS Button"
```

I also have a defined Menu Toggle Gamepad Combo and it works as expected, i.e. it can be pressed in game to enter the menu, and then pressed again in the menu to return to the game.

Reproduced with RA 1.8.8 on Linux.

**Comment by bslenul** [comment]
As an additional information: this seems to happen only when you have `Settings > Input > Menu Controls > All Users Control Menu` turned ON: https://www.reddit.com/r/RetroArch/comments/gqynv7/ps_button_not_exiting_menu/.

**Comment by jstolarek** [comment]
Indeed. I had that option turned on and when I turn it off things work as expected.

**Comment by Sanaki** [comment]
Now that I'm properly awake, I've confirmed this was my issue as well. Never would have thought that option was involved.

**Comment by twinaphex** [comment]
We might have to get rid of that option and reimplement it at a later point in time. It was never properly implemented, leading to the issues you see now before you.

**Comment by zoltanvb** [comment]
This works nowadays, menu button on controller can be used to enter and leave menu even when "All users control menu" is enabled.
